Source/engine/path.cpp

```diff
@@ -162,7 +162,7 @@ int ReconstructPath(const ExploredNodes &explored, PointT dest, int8_t *path, si
 		if (it->second.g == 0) break; // reached start
 		path[len++] = GetPathDirection(it->second.prev, cur);
 		cur = it->second.prev;
-		if (len == maxPathLength) {
+		if (len > maxPathLength) {

```

Comment:
It'd be better to keep this check as is but move it to just before where write to `path` to avoid writing OOB.

Reply:
You are absolutely correct.